joinservice/internal/kubernetes/kubernetes.go

```diff
@@ -70,6 +80,30 @@ func (c *Client) CreateConfigMap(ctx context.Context, configMap corev1.ConfigMap
 	return nil
 }
 
+// AddNodeToJoiningNodes adds the provided node as a joining node CRD.
+func (c *Client) AddNodeToJoiningNodes(ctx context.Context, nodeName string, componentsHash string) error {
+	joiningNodeResource := schema.GroupVersionResource{Group: "update.edgeless.systems", Version: "v1alpha1", Resource: "joiningnodes"}
+
+	joiningNode := &unstructured.Unstructured{}
+	joiningNode.SetUnstructuredContent(map[string]any{
+		"apiVersion": "update.edgeless.systems/v1alpha1",
+		"kind":       "JoiningNode",
+		"metadata": map[string]any{
+			"name": nodeName,
+		},
+		"spec": map[string]any{
+			"name":           nodeName,
+			"componentshash": componentsHash,
+		},
+	})
+
+	_, err := c.dynClient.Resource(joiningNodeResource).Create(ctx, joiningNode, metav1.CreateOptions{})
+	if err != nil {
+		return fmt.Errorf("failed to create joining node: %w", err)
+	}
+	return nil
+}
+
```

Comment:
I think the logic here should be adapted to update the resource if it already exists to ensure we don't get stuck here (could happen if we accidentally issue tickets twice).